ouroboros-network/src/Ouroboros/Network/StrictFingerTree.hs

```diff
@@ -0,0 +1,264 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+-- | Strict variants of 'FingerTree' operations.
+--
+-- TODO: Move this to @cardano-prelude@ as a new module, @Data.FingerTree.Strict@.
+module Ouroboros.Network.StrictFingerTree
```

Comment:
For me it would make more sense to name this module as `Data.Fingertree.Strict`.